src/i_winmusic.c

```diff
@@ -195,41 +196,36 @@ static void CALLBACK MidiStreamProc(HMIDIOUT hMidi, UINT uMsg,
     }
 }
 
-// Allocates the buffer and prepares the MIDI header. Set during initialization
-// by the main thread. BUFFER_INITIAL_SIZE should be large enough to avoid
-// reallocation by the MIDI thread during playback, due to a known memory bug
-// with midiOutUnprepareHeader() (detected by ASan). The calling thread must
-// have exclusive access to the shared resources in this function.
+// Unprepare MIDI header. The calling thread must have exclusive access to the
+// shared resources in this function.
 
-static void AllocateBuffer(const unsigned int size)
+static void UnprepareHeader(void)
```

Comment:
Should this function signal failure to the caller? What should happen if `midiOutUnprepareHeader` fails? What happens with ASAN enabled? I think in that case prepare is called without ever un-preparing. Is that safe?